python_modules/libraries/dagster-dg/dagster_dg/context.py

```diff
@@ -898,3 +902,27 @@ def _get_dg_pypi_version_info(context: DgContext) -> Optional[DgPyPiVersionInfo]
                 context.config.cli.suppress_warnings,
             )
     return version_info
+
+
+def _validate_dagster_dg_and_dagster_version_compatibility(context: DgContext) -> None:
+    dagster_dg_version = Version(__version__)
+    dagster_version = context.dagster_version
+    minimum_dagster_dg_version = Version(library_version_from_core_version(str(dagster_version)))
+    if dagster_dg_version < minimum_dagster_dg_version:
+        exit_with_error(
+            textwrap.dedent(f"""
+            Current `dg` version is incompatible with `dagster-components` version in the resolved environment.
```

Comment:
Can we drop the reference to `dagster-components` CLI entirely, or at least in the preamble? That feels like an implementation detail and IMO reads at first like it refers to a separate `dagster-components` package.
> Current dg version is incompatible with dagster version in the resolved environment. `dg `communicates across a process boundary with `dagster`. Found versions:

(Maybe this is a reason to rename the `dagster-components` entrypoint entirely)